api/rest/client.go

```diff
@@ -0,0 +1,158 @@
+package rest
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/sirupsen/logrus"
+)
+
+const bundlesEndpoint = "/system/health/v1/diagnostics"
+
+// Client is an interface that can talk with dcos-diagnostics REST API and manipulate remote bundles
+type Client interface {
+	// CreateBundle requests the given node to start a bundle creation process with that is identified by the given ID
+	CreateBundle(ctx context.Context, node string, ID string) (*Bundle, error)
```

Comment:
I haven't spot it before. Why do we have pointer here? It's a small struct so IMO it should be a value